File: src/listas/basica/vPython/Ejemplo_mugre.py
from List_mugre import List

def view_line(nephews):
    print("LA FILA" + "-" * 18)
    for nephew in nephews.list_all():
        print(nephew)
    print("-" * 25)

def main():
    nephews = List()

    view_line(nephews)
    print(f"Esta vacia? {nephews.is_empty()}")

    nephews.insert_end("Juanito")
    nephews.insert_end("Pepito")
    nephews.insert_end("Anita")
    print(f"Esta vacia? {nephews.is_empty()}")

    view_line(nephews)

    nephews.insert_end("Luisito")
    print(f"Tamaño: {nephews.size()}")

    nephews.delete()
    print(f"Tamaño: {nephews.size()}")

    view_line(nephews)

if __name__ == "__main__":
    main()

File: production/eda_23_24/listas/basica/vJavascript/Ejemplo_mugre.js
const List = require('./List_mugre');

function viewLine(list) {
    const array = list.listAll();
    console.log("LA FILA" + "-".repeat(18));
    array.forEach(sobrino => {
        console.log(sobrino);
    });
    console.log("-".repeat(25));
}

const nephews = new List();

viewLine(nephews);
console.log("Esta vacia? " + nephews.isEmpty());

nephews.insertEnd("Juanito");
nephews.insertEnd("Pepito");
nephews.insertEnd("Anita");

console.log("Esta vacia? " + nephews.isEmpty());

viewLine(nephews);

nephews.insertEnd("Luisito");

console.log("Tamaño: " + nephews.size());

nephews.delete();
console.log("Tamaño: " + nephews.size());

viewLine(nephews);
